packages/apollo-server-core/src/ApolloServer.ts

```diff
@@ -380,7 +383,10 @@ const typeDefs = gql\`${startSchema}\`
     try {
       context =
         typeof this.context === 'function'
-          ? await this.context({ req: request })
+          ? await this.context({
+              req: request,
+              cache: this.requestOptions.cache as KeyValueCache,
```

Comment:
Why is cache special and deserving of being one of the few keys passed to the context function?